src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -1816,48 +1812,85 @@ int D4NFilterObject::D4NFilterDeleteOp::delete_obj(const DoutPrefixProvider* dpp
   std::string head_oid_in_cache;
   rgw::d4n::CacheBlock block;
   if (!source->check_head_exists_in_cache_get_oid(dpp, head_oid_in_cache, attrs, block, y) && !block.deleteMarker) {
```

Comment:
even here you must add a comment that check_head_exists_in_cache_get_oid will return false if the block exists in cache but is a delete marker